apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/modules/components/OpenEditionMetadata.tsx

```diff
@@ -0,0 +1,220 @@
+    async (values: SetSharedMetadataFormValues) => {
+      if (!ownerAccount) {
+        throw new Error("Not an Owner account");
+      }
+
+      const setSharedMetadataTx = OpenEditionMetadataERC721.setSharedMetadata({
+        contract,
+        metadata: {
+          name: values.name,
+          description: values.description,
+          imageURI: values.imageUri || "",
+          animationURI: values.animationUri || "",
+        },
+      });
+
+      await sendAndConfirmTransaction({
+        transaction: setSharedMetadataTx,
+        account: ownerAccount,
+      });
+    },
+    [contract, ownerAccount],
+  );
+
+  return (
+    <OpenEditionMetadataModuleUI
+      {...props}
+      setSharedMetadata={setSharedMetadata}
+      isOwnerAccount={!!props.ownerAccount}
+    />
+  );
+}
+
+export function OpenEditionMetadataModuleUI(
+  props: Omit<ModuleCardUIProps, "children" | "updateButton"> & {
+    isOwnerAccount: boolean;
+    setSharedMetadata: (values: SetSharedMetadataFormValues) => Promise<void>;
+  },
+) {
+  return (
+    <ModuleCardUI {...props}>
+      <div className="h-1" />
+      <Accordion type="single" collapsible className="-mx-1">
+        <AccordionItem value="metadata" className="border-none">
+          <AccordionTrigger className="border-border border-t px-1">
+            Set Shared Metadata
+          </AccordionTrigger>
+          <AccordionContent className="px-1">
+            {props.isOwnerAccount && (
+              <SetSharedMetadataSection
+                setSharedMetadata={props.setSharedMetadata}
+              />
+            )}
+            {!props.isOwnerAccount && (
+              <Alert variant="info">
+                <CircleAlertIcon className="size-5" />
+                <AlertTitle>
+                  You don't have permission to set shared metadata on this
+                  contract
+                </AlertTitle>
+              </Alert>
+            )}
+          </AccordionContent>
+        </AccordionItem>
+      </Accordion>
+    </ModuleCardUI>
+  );
+}
+
+function SetSharedMetadataSection(props: {
+  setSharedMetadata: (values: SetSharedMetadataFormValues) => Promise<void>;
+}) {
+  const form = useForm<SetSharedMetadataFormValues>({
+    resolver: zodResolver(formSchema),
+    values: {
+      name: "",
+      description: "",
+      imageUri: "",
+      animationUri: "",
+    },
+    reValidateMode: "onChange",
+  });
+
+  const setSharedMetadataMutation = useMutation({
+    mutationFn: props.setSharedMetadata,
+  });
+
+  const onSubmit = async () => {
+    const promise = setSharedMetadataMutation.mutateAsync(form.getValues());
+    toast.promise(promise, {
+      success: "Successfully set shared metadata",
+      error: (error) => `Failed to set shared metadata: ${error}`,
+    });
+  };
+
+  return (
+    <Form {...form}>
+      <form onSubmit={form.handleSubmit(onSubmit)}>
+        <div className="flex flex-col gap-6">
+          <div className="flex gap-4">
```

Comment:
use the `grid grid-cols-1 lg:grid-cols-2` on parent element instead of flex on parent and flex-1 on children elements